### ISSUE TITLE
Incorrect Error Message in RGBDInageFactory.cpp

### ISSUE THREAD

**Issue by redhermes** [isl-org/Open3D#6236]
### Checklist

- [X] I have searched for [similar issues](https://github.com/isl-org/Open3D/issues).
- [X] For Python issues, I have tested with the [latest development wheel](http://www.open3d.org/docs/latest/getting_started.html#development-version-pip).
- [X] I have checked the [release documentation](http://www.open3d.org/docs/release/) and the [latest documentation](http://www.open3d.org/docs/latest/) (for `master` branch).

### Describe the issue

I have been trying to determine what was wrong with my .png file due to the error message "Unsupported image format." in RGBDImage::CreateFromColorAndDepth (line 21). In fact, the problem is not an unsupported image format, but the color and depth images were not the same size.

### Steps to reproduce the bug

```python
Call CreateFromColorAndDepth() with two file of different sizes.
```

### Error message

RuntimeError: [1;31m[Open3D Error] (static std::shared_ptr<open3d::geometry::RGBDImage> open3d::geometry::RGBDImage::CreateFromColorAndDepth(const open3d::geometry::Image&, const open3d::geometry::Image&, double, double, bool)) /root/Open3D/cpp/open3d/geometry/RGBDImageFactory.cpp:21: Unsupported image format.
[0;m

### Expected behavior

An error message that indicates the color and depth file are not the same size.

### Open3D, Python and System information

```markdown
Should not matter but:
Open3D 0.7.0, Python 3.10 Fedora Linux 38
```

### Additional information

Hate to tell you how much time I wasted on this...

**Comment by theNded** [comment]
Thanks for pointing out the issue. I will make a PR and fix this.

**Comment by theNded** [comment]
Closed via #6305
